storage/src/test/java/tech/pegasys/teku/storage/server/AbstractDatabaseTest.java

```diff
@@ -495,7 +495,19 @@ public void handleFinalizationWhenCacheLimitsExceeded() {
     assertThat(tx.commit()).isCompleted();
 
     // All finalized blocks and states should be available
-    assertFinalizedBlocksAndStatesAvailable(newBlocks);
+    final List<SignedBeaconBlock> expectedFinalizedBlocks =
+        newBlocks.stream().map(SignedBlockAndState::getBlock).collect(toList());
+    final Map<Bytes32, BeaconState> expectedFinalizedStates =
+        newBlocks.stream()
+            .filter(
+                blockAndState ->
+                    blockAndState
+                        .getSlot()
+                        .equals(compute_start_slot_at_epoch(blockAndState.getSlot())))
```

Comment:
Why are we filtering for epoch boundary states?  Is this because we're storing hot states at every epoch boundary?

Reply:
Probably worth a comment

Reply:
Yes the test configuration is setup to store a hot state every epoch. I'll add a comment.